lang/java/reef-checkpoint/src/main/java/org/apache/reef/io/checkpoint/RandomNameCNS.java

```diff
@@ -26,20 +26,24 @@
 import javax.inject.Inject;
 
 /**
- * Simple naming service that generates a random checkpoint name.
+ * A naming service that generates a random checkpoint name by appending a random alphanumeric string (suffix)
+ * of a given length to a user-supplied prefix string.
  */
 public class RandomNameCNS implements CheckpointNamingService {
 
   private final String prefix;
+  private final int lengthOfRandomSuffix;
 
   @Inject
-  public RandomNameCNS(@Parameter(PREFIX.class) final String prefix) {
+  public RandomNameCNS(@Parameter(PREFIX.class) final String prefix,
```

Comment:
Technically, this violates our "don't break public APIs" contract. Maybe keep the old constructor and use a default for `lengthOfRandomSuffix`? Also, you can make the new constructor `private`, as it is meant for Tang only.